web/src/constants.ts

```diff
@@ -4,7 +4,126 @@ export const METADATA_URL = "https://raw.githubusercontent.com/homarr-labs/dashb
+export const websiteFullTitle = `${SITE_NAME} ${TITLE_SEPARATOR} Free, Curated Icons for Apps & Services ${TITLE_SEPARATOR} ${SITE_TAGLINE}`
+
+// Various keyword sets for different pages
+export const DEFAULT_KEYWORDS = [
+	"dashboard icons",
+	"app icons",
+	"service icons",
+	"curated icons",
+	"free icons",
+	"SVG icons",
+	"web dashboard",
+	"app directory",
+]
+
+export const BROWSE_KEYWORDS = [
+	"browse icons",
+	"search icons",
+	"download icons",
+	"minimal icons",
+	"dashboard design",
+	"UI icons",
+	...DEFAULT_KEYWORDS,
+]
+
+// Add format-specific keywords
+export const ICON_DETAIL_KEYWORDS = (iconName: string): string[] => [
+	`${iconName} icon`, // e.g., "Homarr icon"
+	`${iconName} logo`, // e.g., "Homarr logo"
+	`${iconName} svg icon`, // e.g., "Homarr svg icon"
+	`${iconName} png icon`, // e.g., "Homarr png icon"
+	`${iconName} webp icon`, // e.g., "Homarr webp icon"
+	`${iconName} download`, // e.g., "Homarr download"
+	`${iconName} dashboard icon`, // e.g., "Homarr dashboard icon"
+	...DEFAULT_KEYWORDS,
+]
+
+// Core structured data for the website (JSON-LD)
+export const getWebsiteSchema = (totalIcons: number) => ({
+	"@context": "https://schema.org",
+	"@type": "WebSite",
+	name: SITE_NAME,
+	url: WEB_URL,
+	description: getDescription(totalIcons),
+	potentialAction: {
+		"@type": "SearchAction",
+		target: {
+			"@type": "EntryPoint",
+			urlTemplate: `${WEB_URL}/icons?q={search_term_string}`,
+		},
+		"query-input": "required name=search_term_string",
+	},
+	slogan: SITE_TAGLINE,
+})
+
+// Organization schema
+export const ORGANIZATION_SCHEMA = {
+	"@context": "https://schema.org",
+	"@type": "Organization",
+	name: ORGANIZATION_NAME,
+	url: `https://github.com/${REPO_NAME}`,
+	logo: `${WEB_URL}/og-image.png`,
+	sameAs: [`https://github.com/${REPO_NAME}`, "https://homarr.dev"],
+	slogan: SITE_TAGLINE,
+}
+
+// Social media
+export const GITHUB_URL = `https://github.com/${REPO_NAME}`
+
+// Image schemas
+export const getIconSchema = (
+	iconName: string,
+	iconId: string,
+	authorName: string,
+	authorUrl: string,
+	updateDate: string,
+	totalIcons: number,
+) => ({
+	"@context": "https://schema.org",
+	"@type": "ImageObject",
+	name: `${iconName} Icon`,
+	description: getIconDescription(iconName, totalIcons),
+	contentUrl: `${BASE_URL}/png/${iconId}.png`,
+	thumbnailUrl: `${BASE_URL}/png/${iconId}.png`,
+	uploadDate: updateDate,
+	author: {
+		"@type": "Person",
+		name: authorName,
+		url: authorUrl,
+	},
+	encodingFormat: ["image/png", "image/svg+xml", "image/webp"],
+	contentSize: "Variable",
+	representativeOfPage: true,
+	creditText: `Icon contributed by ${authorName} to the ${SITE_NAME} collection by ${ORGANIZATION_NAME}`,
+	embedUrl: `${WEB_URL}/icons/${iconId}`,
+})
 
-export const websiteTitle = "Free Dashboard Icons - Download High-Quality UI & App Icons"
+// OpenGraph defaults
+export const DEFAULT_OG_IMAGE = {
```

Comment:
This should just be in the layout, in the nextjs example they even show that you can have multiple images by using like `parent.ogImages` or something similar